### PR TITLE
perf(core): move origin type map to module level in `function_calling.py`

### PR DESCRIPTION
**Pull request by Rudra-Tiwari-codes**
Moves `_ORIGIN_MAP` dict from inside `_py_38_safe_origin()` to module level constant. This avoids dict allocation on every function call, reducing garbage collection pressure during frequent tool conversions.

The function is called during typed dict to pydantic model conversion which happens during tool binding and invocation - a hot path in LangChain.

**Testing:** `make lint` passes

### CODE REVIEW

**Comment by Rudra-Tiwari-codes** [comment]
The failing test test_uuid_utils.py::test_monotonicity is unrelated to the PR...it's a flaky test that fails intermittently due to race conditions in parallel UUID generation. Re-running CI...

File: libs/core/langchain_core/utils/function_calling.py [hunk 1]
```diff
@@ -48,6 +48,22 @@
     "bool": "boolean",
 }
 
+# Precomputed origin type mapping for _py_38_safe_origin
+# Avoids dict creation on every function call
```
Comment (mdrxy) [inline libs/core/langchain_core/utils/function_calling.py]:
```suggestion
```

**Review by mdrxy — APPROVED** [review]
Thank you!

**Comment by fransafu** [comment]
@mdrxy 

**Warning:** This author is forking multiple ML projects such as google-deepmind/alphafold, ml-explore/mlx, openai/CLIP, pytorch/pytorch, tensorflow/tensorflow, anthropics/claude-code, vllm-project/vllm, and others, adding minimal "contributions" (often for tests or miscellaneous changes) without proper validation. A review of their commits shows mostly local implementations of TODOs copied from existing projects, with little to no substantive review or testing.

So far, this author has forked **41 repositories** following the same pattern. Be careful when accepting this PR. It’s also concerning how this author is able to submit PRs across four repositories in the same day, each requiring large context, which strongly suggests a highly automated workflow.

**Comment by Rudra-Tiwari-codes** [comment]
Thank you for the feedback regarding my recent activity. I am a student and I have been using these smaller tasks as a way to familiarize myself with the architecture of various codebases. I was not aware that submitting multiple minor pull requests was considered disruptive to the maintainer workflow or seen as contribution padding. I appreciate the correction and will pay much closer attention to the impact of my work moving forward. I am closing this pull request now to focus on delivering more substantive technical contributions that provide genuine value to the community.

**Review by tayn58188-png — COMMENTED** [review]
B
